src/DocumentFormat.OpenXml/Packaging/OpenSettings.cs

```diff
@@ -61,5 +67,10 @@ public MarkupCompatibilityProcessSettings MarkupCompatibilityProcessSettings
         /// This property allows you to mitigate denial of service attacks where the attacker submits a package with an extremely large Open XML part. By limiting the size of the part, you can detect the attack and recover reliably.
         /// </remarks>
         public long MaxCharactersInPart { get; set; }
+
+        /// <summary>
+        /// Gets or sets a delegate that is used to create a handler to rewrite relationships that are malformed. On platforms after .NET 4.5, <see cref="Uri"/> parsing will fail on malformed strings.
+        /// </summary>
+        public Func<OpenXmlPackage, RelationshipErrorHandler> RelationshipErrorHandlerFactory { get; set; }
```

Comment:
I think using the interface will be better than using Func

Reply:
Can you explain a bit more what you mean by this? This simply a factory to get a handler. How can an interface help here?

Reply:
@twsouthwick   Thank you. This is just the habitual use of factory design patterns.

An optimization point is as the following code:

       openSettings.RelationshipErrorHandlerFactory  += Foo1;
       openSettings.RelationshipErrorHandlerFactory  += Foo2;

Reply:
Not sure why this would be an optimization. I'm not personally a fan of events as they are difficult to compose (you kind of can, though, via System.Reactive with some weird hoops initially). If needed, you could create a `RelationshipErrorHandler` that could compose multiple handlers together that hand off to the next one if needed.